extensions/interactions/GraphInput/directives/GraphInput.js

```diff
@@ -45,49 +48,63 @@ oppia.directive('oppiaInteractiveGraphInput', [
             isWeighted: false,
             isLabeled: false
           };
-
           $scope.submitGraph = function() {
             // Here, angular.copy is needed to strip $$hashkey from the graph.
             $scope.onSubmit({
               answer: angular.copy($scope.graph),
               rulesService: graphInputRulesService
             });
           };
+          $scope.interactionIsActive = ($scope.getLastAnswer() === null);
+          $scope.$on(EVENT_NEW_CARD_AVAILABLE, function(evt, data) {
+            $scope.interactionIsActive = false;
+            $scope.canAddVertex = false;
+            $scope.canDeleteVertex = false;
+            $scope.canEditVertexLabel = false;
+            $scope.canMoveVertex = false;
+            $scope.canAddEdge = false;
+            $scope.canDeleteEdge = false;
+            $scope.canEditEdgeWeight = false;
+          });
           $scope.resetGraph = function() {
-            updateGraphFromJSON($attrs.graphWithValue);
+            var newGraph =
+              HtmlEscaperService.escapedJsonToObj($attrs.graphWithValue);
```

Comment:
Generally, prefer breaking after "(" rather than after "=", i.e.:

    var newGraph = HtmlEscaperService.escapedJsonToObj(
      $attrs.graphWithValue);

Reply:
Done